### ISSUE TITLE
B008: mark builtins.slice() as immutable

### ISSUE THREAD

**Issue by anabelle2001** [astral-sh/ruff#21565]
# Example
```py
def a(b=float("inf")): ... # True Negative
def c(d=slice(None)): ... # False Positive, as slice() is immutable
```

There is also the semi-related corner case of `def e(f: slice = dict()): ...`. This is clearly erroneous code, but it is incorrectly flagged as violating B008, despite the docs saying "Parameters with immutable type annotations will be ignored by this rule". (e.g., `def g(h: int = dict()): ...`, does not trigger B008)

see also: [docs.python.org > builtins > slice](https://docs.python.org/3/library/functions.html#slice)

**Comment by ntBre** [comment]
Thanks! It makes sense to me to update this list to include slice:

https://github.com/astral-sh/ruff/blob/f04a18d8f82afc8d60af053d1b121f32d719b454/crates/ruff_python_stdlib/src/typing.rs#L329

based on the linked docs:

> Slice objects have read-only data attributes start, stop, and step which merely return the argument values (or their default). They have no other explicit functionality

I think that should also resolve the annotation case.
